src/WebJobs.Extensions.DurableTask/DefaultPlatformInformationProvider.cs

```diff
@@ -49,6 +44,60 @@ public bool InLinuxAppService()
             return inLinuxDedicated;
         }
 
+        public OperatingSystem GetOperatingSystem()
+        {
+            if (this.InLinuxConsumption() || this.InLinuxAppService())
+            {
+                return OperatingSystem.Linux;
+            }
+
+            return OperatingSystem.Windows;
+        }
+
+        public AppServicePlan GetAppSevicePlan()
```

Comment:
The name "AppServicePlan" is misleading because it has a very specific meaning in the context of Azure (it refers to a _name_ of a plan, and excludes Consumption and ElasticPremium plan types, which are technically distinct from "App Service plans"). Consider "PlanType" instead for both the name of this enum and the name of this method.